### PR TITLE
React to async topic feedback

### LINKED ISSUE #2289: Add cross-references to conceptual async docs + react to feedback
Implement feedback in https://github.com/dotnet/docs/pull/2284.

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #2289

### CODE REVIEW

File: docs/standard/async.md [hunk 1]
```diff
@@ -30,9 +30,8 @@ Async code has the following characteritics:
 * Handles more server requests by yielding threads to handle more requests while waiting for I/O requests to return.
```
Comment (rpetrusha) [inline docs/standard/async.md]: Line 28 (earlier): characteritics --> characteristics
We're stressing how easy async code is, but it's only the task-based async pattern with compiler support that's easy. In line 26, I'd change "Async APIs and the language level..." to "Task-based async APIs and the language-level..."

File: docs/standard/async.md [hunk 2]
```diff
@@ -30,9 +30,8 @@ Async code has the following characteritics:
...
 * Enables UIs to be more responsive by yielding threads to UI interaction while waiting for I/O requests and by transitioning long-running work to other CPU cores.
 * Many of the newer .NET APIs are asynchronous.
-* It’s super easy to write async code in .NET!
```
Comment (rpetrusha) [inline docs/standard/async.md]: Delete "super".

File: docs/standard/async.md [hunk 3]
```diff
@@ -30,9 +30,8 @@ Async code has the following characteritics:
...
+* It's super easy to write async code in .NET!
 
-## What’s next?
+## What's next?
 
 For a deep dive into async concepts and programming, see [Async in depth](async-in-depth.md).
```
Comment (rpetrusha) [inline docs/standard/async.md]: Also add a link to [Task-based asynchronous programming](parallel-programming/task-based-asynchronous-programming.md).

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thanks for doing this so quickly, @GuardRex. It looks good. Just a few minor changes/comments.

**Comment by GuardRex** [comment]
@rpetrusha Changes made. :+1:

**Review by rpetrusha — APPROVED** [review]
@GuardRex, LGTM. We can merge when the build completes.

**Comment by GuardRex** [comment]
@rpetrusha I needed to fix one link. While I was looking at the review docs, I felt that having the added cross-reference links in the *docs/standard/asynchronous-programming-patterns/index.md* without descriptions didn't look good. I reverted the existing three links at the bottom of that topic (in a table format) back into a *Related topics* section and added the three new cross-reference links in a traditional *See also* section (where they don't require a description). If that's ok with you, then this should be ok for final :eyes:+:shipit: in the morning.

**Review by rpetrusha — COMMENTED** [review]
Sorry I didn't notice this earlier, @GuardRex, but can you change the link in line 32 of async-in-depth.md to point to the docs topic [Task-based asynchronous pattern](https://docs.microsoft.com/en-us/dotnet/standard/asynchronous-programming-patterns/task-based-asynchronous-pattern-tap) rather than the MSDN topic?

**Pull request merged**